dask/dataframe/io.py

```diff
@@ -447,27 +447,42 @@ def to_hdf(df, path_or_buf, key, mode='a', append=False, complevel=0,
 
     # when writing to the same file in multiple tasks we should lock
     if lock is None:
-        if not single_node:
+        if not single_node or not single_file:
             lock = True
         else:
             lock = False
 
-    # if get not specified default to sync if we're writing to a single entity
-    # if we're writing to multiple files, default to multiprocessing
-    # by default hdf5 is not thread safe, so threaded is not an option
-    if get is None:
-        if not single_file:
-            get = dask.multiprocessing.get
-        else:
-            get = get_sync
+    if lock is True:
+        lock = Lock()
+
+    def _pd_to_hdf(*args, **kwargs):
+        lock = kwargs.pop('lock')
+
+        if lock:
+            go = False
+            while not go:
+                print(lock)
+                go = lock.acquire(blocking=False)
+                print(go)
+            #lock.acquire()
+            print("acquired")
+        try:
+            pd_to_hdf(*args, **kwargs)
+        finally:
+            if lock:
+                lock.release()
+                print("released")
+
+        return None
```

Comment:
If possible this function should be moved outside of the function so that it is easier to serialize.  We should probably also remove the go and print logic.